lib/libproc/proc_util.c

```diff
@@ -180,8 +181,20 @@ proc_getwstat(struct proc_handle *phdl)
 char *
 proc_signame(int sig, char *name, size_t namesz)
 {
-
-	strlcpy(name, strsignal(sig), namesz);
+	char buf[SIG2STR_MAX + 3];
+	size_t len;
+
+	/* sig2str() omits the leading "SIG" */
+	(void)strlcpy(buf, "SIG", sizeof(buf));
+
+	if (sig2str(sig, buf + 3) == 0) {
+		(void)strlcpy(name, buf, namesz);
+	} else {
+		len = snprintf(name, namesz, "SIG#%d", sig);
+		/* ensure null-termination */
```

Comment:
That's not needed as `snprintf()` already takes care of it. Since libproc already has ATF tests, I suggest you add a test case which verifies that `proc_signame()` not only produces the correct result but also correctly terminates the string if the buffer is too short.

Reply:
> That's not needed as `snprintf()` already takes care of it. Since libproc already has ATF tests, I suggest you add a test case which verifies that `proc_signame()` not only produces the correct result but also correctly terminates the string if the buffer is too short.

You're right. Fixed it.  I think this function is not worth creating a new test since it only has 2 consumers.